src/modules/scraper/adapter/messaging/FillEventsConsumer.ts

```diff
@@ -19,56 +19,91 @@ export class FillEventsConsumer {
-      });
-      const olderDeposit = await this.depositRepository.findOne({
-        where: { sourceChainId: originChainId, depositId: LessThan(depositId) },
-        order: { depositId: "desc" },
-      });
-
-      if (newerDeposit && olderDeposit) {
-        const now = new Date();
-        const diffMinutes = Math.round((now.getTime() - newerDeposit.createdAt.getTime()) / 60000);
-        if (diffMinutes >= 20) {
-          await this.scraperQueuesService.publishMessage<BlocksEventsQueueMessage>(ScraperQueue.BlocksEvents, {
-            chainId: originChainId,
-            from: olderDeposit.blockNumber,
-            to: newerDeposit.blockNumber,
-          });
-        }
-      }
+      await this.tryToRefetchDepositEvents(job.data);
       throw new Error("Deposit not found in db");
     }
 
-    const fillTxIndex = deposit.fillTxs.findIndex(
-      (fillTx) => fillTx.hash === transactionHash && fillTx.totalFilledAmount === totalFilledAmount,
-    );
-
-    if (fillTxIndex !== -1) {
+    if (this.fillTxAlreadyProcessed(deposit, job.data)) {
       this.logger.warn("Fill event already processed");
       return;
     }
 
-    deposit.realizedLpFeePct = new BigNumber(deposit.realizedLpFeePct).plus(realizedLpFeePct).toString();
-    // for referrals, the realized lp fee should be capped at 0.12%
-    const maxRealizedLpFeePct = new BigNumber(10).pow(18).times(0.0012);
-    deposit.realizedLpFeePctCapped = BigNumber.min(deposit.realizedLpFeePct, maxRealizedLpFeePct).toString();
+    deposit.fillTxs = [
+      ...deposit.fillTxs,
+      { fillAmount, hash: transactionHash, isSlowRelay, realizedLpFeePct, totalFilledAmount },
+    ];
+    const bridgeFeePct = this.computeBridgeFee(deposit, job.data);
 
     if (new BigNumber(deposit.filled).lt(totalFilledAmount)) {
       deposit.filled = totalFilledAmount;
     }
 
-    deposit.fillTxs = [...deposit.fillTxs, { hash: transactionHash, totalFilledAmount }];
     deposit.status = new BigNumber(deposit.amount).eq(deposit.filled) ? "filled" : "pending";
+    deposit.bridgeFeePct = bridgeFeePct.toString();
 
     await this.depositRepository.save(deposit);
   }
 
+  private computeBridgeFee(deposit: Deposit, fill: FillEventsQueueMessage) {
+    const maxBridgeFeePct = new BigNumber(10).pow(18).times(0.0012);
+    const validFills = deposit.fillTxs.filter((fill) => !fill.isSlowRelay); // all fills associated with a deposit that are NOT slow fills
+    const relayerFeePctChargedToUser = validFills.reduce((cumulativeFee, fill) => {
+      const relayerFee = new BigNumber(fill.fillAmount).multipliedBy(deposit.depositRelayerFeePct);
+      return relayerFee.plus(cumulativeFee);
+    }, new BigNumber(0));
+    const blendedRelayerFeePct = relayerFeePctChargedToUser.dividedBy(deposit.amount).decimalPlaces(0, 1);
+    const bridgeFeePct = blendedRelayerFeePct.plus(fill.realizedLpFeePct);
+    const bridgeFeePctCapped = BigNumber.min(bridgeFeePct, maxBridgeFeePct);
+
+    return bridgeFeePctCapped;
+  }
```

Comment:
this is the important code to be reviewed. This is how the bridge fee is computed